ui/src/controller/track_decider.ts

```diff
@@ -219,18 +249,32 @@ class TrackDecider {
       args: Partial<{
         upid: number|null,
         processName: string|null,
-        numThreads: number|null,
+        totalThreads: number|null,
+        idleThreads: number|null,
+        nullThreads: number|null,
       }>): string | undefined {
-    const {upid, processName, numThreads} = args;
+    const {upid, processName, totalThreads, idleThreads, nullThreads} = args;
 
     const hasProcessName = !!processName && upid !== undefined && upid !== null;
-    const hasNumThreads = numThreads !== undefined && numThreads !== null;
+    const hasTotalThreads = totalThreads !== undefined && totalThreads !== null;
+    const hasIdleThreads = idleThreads !== undefined && idleThreads !== null &&
+      idleThreads > 0;
+    const hasNullThreads = nullThreads !== undefined && nullThreads !== null &&
+      nullThreads > 0;
+    const count = (n: number, single: string, plural = single + 's') => n === 1 ?
+      `1 ${single}` :
+      `${n} ${plural}`;
 
     let suffix = '';
     if (hasProcessName) {
       suffix = `Process: ${processName} [${upid}]`;
-    } else if (hasNumThreads) {
-      suffix = numThreads === 1 ? '1 thread' : `${numThreads} threads`;
+    } else if (hasIdleThreads && hasTotalThreads) {
+      const allIdle = idleThreads + (nullThreads ?? 0);
+      suffix = `${count(totalThreads, 'thread')} of which ${count(allIdle, 'is', 'are')} idle.`;
```

Comment:
> nit: for the pedantic. Avoid misrepresenting threads with no data as idle ones. Mention both possibilities.

The request in the original issue is to make parity with AGI, which treats threads that generated no data as idle, which makes sense because if they weren't idle, then they would have generated some data. The difference in Perfetto is that Perfetto doesn't create tracks for them.

Reply:
Maybe I'm missing something here, but if we're going to lump idle and no-data threads in the same bucket, then why are we making a differentiation between the two at line 273 and 275?  

I.e., if we have 10 idle threads and 5 threads with no data, we say that we have 15 idle threads (line 273). If we have 0 idle threads and 5 threads with no data, we say we have 5 no-data threads (line 275). That seems a bit inconsistent to me. BTW, I'm fine considering <1% threads and no-data threads both as idle, but then we shouldn't have lines 274-275, and lines 271-273 should cover the scenario where we have zero idle threads and some number of no-data threads.

Reply:
Yeah. I'm trying to be clear in the explanation to the user of why there's a count of things (processes or threads) that are idle when that count is more than what the UI shows. So that if the user actually counts things in the UI, they won't add up, and it would be confusing if not explained.

I'll try to find a more obvious way of actually constructing all of this explanatory text.